### PR TITLE
Ansible remediation and tests for audit_rules_immutable

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:
Added ansible remediation and tests for audit_rules_immutable.

#### Rationale:

CIS productization

### CODE REVIEW

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  with_items: "{{ files_to_search }}"
+
+- name: insert lines at the end of /etc/audit/audit.rules and /etc/audit/rules.d/immutable.rules
+  blockinfile:
+    path: "{{ item }}"
+    create: True
+    marker: ""
+    block: |+
+      # Set the audit.rules configuration immutable per security requirements
+      # Reboot is required to change audit rules once this setting is applied
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml:40]: When run multiple times, these blocks become repeated.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    recurse: true
+    contains: "-e[\\s]+.*"
+    patterns: "*.rules"
+  register: find_immutable
+
+- name: add found files to the list of files to be searched
+  set_fact:
+    files_to_search: "{{ files_to_search + [item.path] }}"
+  with_items: "{{ find_immutable.files }}"
+  when: find_immutable.matched is defined and find_immutable.matched >= 1
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml:24]: The files are to be edited, not searched.

**Comment by mildas** [comment]
**Findings:**
Rule audit_rules_immutable:
 Ansible remediation newly added.

**Recommended tests to execute:**
 build_product fedora
 test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate ansible --datastream build/ssg-fedora-ds.xml audit_rules_immutable

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml [hunk 3]
```diff
@@ -1,45 +1,51 @@
...
+        files_to_edit: []
+    - name: Add matched files from /etc/audit/rules.d
+      set_fact:
+        files_to_edit: "{{ files_to_edit + [item.path] }}"
+      loop: "{{ find_immutable_rules_d.files }}"
+    - name: Add /etc/audit/audit.rules to the list of files
+      set_fact:
+        files_to_edit: "{{ files_to_edit + ['/etc/audit/audit.rules'] }}"
+      when: find_immutable_audit_rules is defined and find_immutable_audit_rules.matched >= 1
+  when: (find_immutable_rules_d.matched is defined and find_immutable_rules_d.matched >= 1) or (find_immutable_audit_rules.matched is defined and find_immutable_audit_rules.matched >= 1)
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml:34]: There is no need to carefully craft the list of files to edit.

File: linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml [hunk 4]
```diff
@@ -1,45 +1,51 @@
...
 
-- name: remove the config line from /etc/audit/audit.rules and any file in /etc/audit/rules.d directory
+- name: Remove the -e option from all Audit config files
   lineinfile:
     path: "{{ item }}"
     regexp: "-e[\\s]+.*"
     state: absent
-  with_items: "{{ files_to_search }}"
+  loop: "{{ files_to_edit }}"
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/auditd_configure_rules/audit_rules_immutable/ansible/shared.yml:41]: This task can be run over `/etc/audit/audit.rules` and any `*.rules` file in `/etc/audit/rules.d`.

**Comment by mildas** [comment]
**Changes identified:**
Rule audit_rules_immutable:
 Ansible remediation newly added.

**Recommended tests to execute:**
 build_product fedora
 test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate ansible --datastream build/ssg-fedora-ds.xml audit_rules_immutable

**Comment by mildas** [comment]
**Changes identified:**
Rule audit_rules_immutable:
 Ansible remediation newly added.

**Recommended tests to execute:**
 build_product fedora
 test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate ansible --datastream build/ssg-fedora-ds.xml audit_rules_immutable

**Comment by yuumasato** [comment]
@vojtapolasek Thank you for the addition. And thanks for keeping it simple.

**Pull request merged**